drivers: mcux: Convert MCUX drivers to new DT_INST macros

galak opened the pull request:
Convert older DT_INST_ macro use in MCUX drivers to the new
include/devicetree.h DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
drivers: mcux: Convert MCUX drivers to new DT_INST macros

Convert older DT_INST_ macro use in MCUX drivers to the new
include/devicetree.h DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

henrikbrixandersen commented on drivers/adc/adc_mcux_adc12.c:
```diff
@@ -252,26 +253,26 @@ static const struct adc_driver_api mcux_adc12_driver_api = {
 #define TO_ADC12_CLOCK_DIV(val) _DO_CONCAT(kADC12_ClockDivider, val)
 
-#if DT_INST_0_NXP_KINETIS_ADC12
+#if DT_HAS_NODE(DT_INST(0, nxp_kinetis_adc12))
```
Can we use `DT_DRV_COMPAT` definition here?

galak replied:
Introduced `DT_HAS_DRV_INST`, now its `#if DT_HAS_DRV_INST(0)`

henrikbrixandersen commented on drivers/clock_control/clock_control_mcux_pcc.c:
```diff
@@ -65,12 +67,12 @@ static const struct clock_control_driver_api mcux_pcc_api = {
 };
 
-#if defined(DT_INST_0_NXP_KINETIS_PCC_LABEL)
+#if DT_INST_NODE_HAS_PROP(0, label)
```
This should likely be `#if DT_HAS_NODE(DT_INST(0, nxp_kinetis_pcc))`.

galak replied:
why?  do we different compats for PCC?

galak replied:
ah, get the point now.  Will change it over from testing `label`

henrikbrixandersen commented on drivers/watchdog/wdt_cmsdk_apb.c:
```diff
@@ -4,6 +4,8 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#define DT_DRV_COMPAT arm_cmsdk_watchdog
```
Did these changes sneak in by mistake? They are not MCUX specific.

galak replied:
Yeah, will remove.

henrikbrixandersen reviewed (approved):
LGTM, thanks.

galak committed:
soc: nxp_kinetis: Convert to new DT_INST macros

Convert older DT_INST_ macro use the new include/devicetree.h
DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
soc: nxp_imx: Convert to new DT_INST macros

Convert older DT_INST_ macro use the new include/devicetree.h
DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
soc: nxp_lpc: Convert to new DT_INST macros

Convert older DT_INST_ macro use the new include/devicetree.h
DT_INST macro APIs.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

mbolivar-nordic reviewed (approved):
Looks sane. I just skimmed these.

vanwinkeljan reviewed (approved):
+1 for display related changes
